src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -64,7 +68,7 @@ public void setUp() {
     private void initResource() {
         when(AbstractDataSetAPI.getInstance()).thenReturn(api);
         when(AbstractDataAccessRequestAPI.getInstance()).thenReturn(dataAccessRequestAPI);
-        resource = new DataSetResource(userService);
+        resource = new DataSetResource(dataSetService, userService);
     }
```

Comment:
For the test, I like to have a very simple test to ensure that the response we get is a 201. Something as simple as 
```
    @Mock
    UriInfo info;

    @Mock
    UriBuilder builder;

    @Test
    public void testCreateDataset() {
        when(builder.build()).thenReturn(URI.create("https://test.domain.org/some/path"));
        when(info.getRequestUriBuilder()).thenReturn(builder);
        initResource();
        Response response = resource.createDataSet(info, "");
        assertEquals(201, response.getStatus());
    }
```
should be enough for us to get a good pattern started so we can layer in other tests that look for failure conditions later on down the line.